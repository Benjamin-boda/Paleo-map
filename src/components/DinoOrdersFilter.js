import React, {useState, useEffect} from "react";
import {Form} from "react-bootstrap";

export const DinoOrdersFilter = ({loaded, dinosauriaOrderArray, setDinosauria}) => {
    const [order, setOrder] = useState("")

    const onOrderChange = (e) => {
        setOrder(e.target.value)
    }

    useEffect(() => {
        if (order !== "") {
            const NoUndefined = dinosauriaOrderArray.filter((dino) => dino.cll !== undefined)
            const orderFiltered = NoUndefined.filter((dino) => dino.cll === order)
            setDinosauria(orderFiltered)
        } else {
            setDinosauria(dinosauriaOrderArray)
        }
         
    }, [order, loaded])

    return (
        
            <Form.Control className="filters__select" as="select" onChange={onOrderChange}>
                <option value="">--Filter by order--</option>
                <option value="Saurischia">Saurischia</option>
                <option value="Ornithischia">Ornithischia</option>
                <option value="Aves">Aves</option>
            </Form.Control>
        
    )
}