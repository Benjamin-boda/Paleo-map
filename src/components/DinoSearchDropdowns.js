import React, {useEffect, useState} from "react";
import {Dropdown} from "react-bootstrap";

export const DinoSearchDropdowns = ({dinosauria, setOpenDropdown, keyword, setIdModal, setModalIsOpen}) => {
    const [dinoSearch, setdinoSearch] = useState([])

    useEffect(() => {
        const NoUndefined = dinosauria.filter((dino) => dino.gnl !== undefined)
        const NoOtherProperties = NoUndefined.map(({gnl, oid, cll}) => ({gnl, oid, cll}))
        const NoDuplicate = NoOtherProperties.filter((dino, index, self) =>
            index === self.findIndex((element) => (
                element["gnl"] === dino["gnl"]
            ))
        )
        const search = NoDuplicate.filter((dino) => dino.gnl.toLowerCase().indexOf(keyword) > -1)
        setdinoSearch(search.slice(0,10))
        
        if (keyword === "") {
            setOpenDropdown(false)
        }
    }, [keyword])

    return (
        <div>
            {dinoSearch.map((dino) => 
                <li className="header__dropdown" key={dino.oid} 
                    onClick={() => {
                        setModalIsOpen(true)
                        setIdModal(dino.oid)
                        setOpenDropdown(false)
                }}>
                    {dino.gnl} - {dino.cll}
                </li>
            )}
            <Dropdown.Divider />
            <li className="header__dropdown" onClick={() => setOpenDropdown(false)} >Close</li>
        </div>
    )
}