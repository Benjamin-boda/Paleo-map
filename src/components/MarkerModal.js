import React, {useState, useEffect} from "react";
import {Modal, Button} from "react-bootstrap";

export const MarkerModal = ({dino, modalIsOpen, setModalIsOpen}) => {
    const [wikiInfos, setWikiInfos] = useState([])
    const [sourceError, setSourceError] = useState(false)
    

    useEffect(() => {
        async function getWikiApi() {
            await fetch(`https://en.wikipedia.org/w/api.php?origin=*&action=query&prop=pageimages&titles=${dino.gnl}&pithumbsize=500&format=json`)
                .then((req) => req.json())
                .then((res) => {
                    const wikiSource = res.query.pages
                    setWikiInfos(Object.values(wikiSource).map(({ thumbnail }) => thumbnail)[0].source)
                    setSourceError(false)
                })
                .catch((err) => setSourceError(true))
        }

        getWikiApi()
    }, [])

    return (
        <Modal aria-labelledby="contained-modal-title-vcenter" centered show={modalIsOpen} onHide={() => setModalIsOpen(!modalIsOpen)}>
            <Modal.Header className="modal__title">{dino.gnl}</Modal.Header>
            <Modal.Body>
                {sourceError ? undefined : <img className="modal__image" src={wikiInfos}/>}
                <p className="modal__text"><span className="modal__text__bold">Location : </span>{dino.ggc}</p>
                <p className="modal__text__bold">Clade : <a className="modal__link" target="_blank" href={`https://en.wikipedia.org/wiki/${dino.cll}`}>{dino.cll}</a></p>
                <p className="modal__text__bold">Details on Wikipedia : <a className="modal__link" target="_blank" href={`https://en.wikipedia.org/wiki/${dino.gnl}`}>{dino.gnl}</a></p>
            </Modal.Body>
            <Modal.Footer>
                <Button variant="secondary" size="sm" onClick={() => setModalIsOpen(!modalIsOpen)}>Close</Button>
            </Modal.Footer>
        </Modal>
    )
}