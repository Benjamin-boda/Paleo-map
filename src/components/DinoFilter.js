import React, { useEffect, useState } from "react";
import {countryList} from "../map/country";
import {DinoOrdersFilter} from "./DinoOrdersFilter";
import {Form} from "react-bootstrap";

export const DinoFilter = ({loaded, continentOrCountry, setContinentOrCountry, dinosauriaOrderArray, setDinosauria}) => {
    const [continentName, setContinentName] = useState(null)
    const [countryFiltered, setCountryFiltered] = useState(countryList)

    useEffect(() => {
        if (continentName === null || continentOrCountry === "") {
            setCountryFiltered(countryList)
        } else {
            setCountryFiltered(countryList.filter((country) => country.Continent_Name === continentName)) 
        }
       
    }, [continentName])
    

    

    const onContinentChange = (e) => {
        setContinentOrCountry(e.target.value)
        setContinentName(e.target.selectedOptions[0].outerText)
    }

    const onCountryChange = (e) => {
        setContinentOrCountry(e.target.value)
    }

    return (
        <div className="filters__background">
            <div className="filters">

                <Form.Control id="continent" className="filters__select" as="select" onChange={onContinentChange}>
                    <option value="">--Filter by continent--</option>
                    <option value="&cc=ATA">Antarctica</option>
                    <option value="&cc=AFR">Africa</option>
                    <option value="&cc=ASI">Asia</option>
                    <option value="&cc=EUR">Europe</option>
                    <option value="&cc=NOA">North America</option>
                    <option value="&cc=OCE">Oceania</option>
                    <option value="&cc=SOA">South America</option>
                </Form.Control>

                <Form.Control className="filters__select" as="select" onChange={onCountryChange}>
                    <option value="">--Filter by country--</option>
                    {countryFiltered.map((country) => 
                        <option key={country.Country_Name + country.Continent_Code} value={`&cc=${country.Two_Letter_Country_Code}`}>
                            {country.Country_Name.includes(",") ? country.Country_Name.substring(0, country.Country_Name.indexOf(",")) : country.Country_Name}
                        </option>   
                    )}
                </Form.Control>

                <DinoOrdersFilter
                    loaded={loaded}
                    dinosauriaOrderArray={dinosauriaOrderArray} 
                    setDinosauria={setDinosauria}
                />
            </div>
        </div>
        
    )
}