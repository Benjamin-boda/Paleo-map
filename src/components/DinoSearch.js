import React, {useState} from "react";
import {InputGroup, FormControl, Dropdown} from "react-bootstrap";
import {DinoSearchDropdowns} from "./DinoSearchDropdowns";

export const DinoSearch = ({dinosauria, loaded, setIdModal, setModalIsOpen}) => {
    const [keyword, setKeyword] = useState("")
    const [openDropdown, setOpenDropdown] = useState(false)

    const onKeywordChange = (e) => {
        const keywordLower = e.target.value
        setKeyword(keywordLower.toLowerCase())
        setOpenDropdown(true)
    }

    return (
        <div className="header">
            <div className="header__content">

                <img className="header__logo" src={require("../images/logo.png").default}/>

                <h1 className="header__text">PALEO MAP</h1>

                <InputGroup 
                    size="lg"
                    className="header__input" 
                    onChange={onKeywordChange}
                    type="text"
                    placeholder="Search a specimen..."
                    value={keyword}
                >
                        <InputGroup.Prepend>
                            <InputGroup.Text className="InputGroup__Text" id="inputGroup-sizing-lg">Search a specimen</InputGroup.Text>
                        </InputGroup.Prepend>
                        <FormControl
                        aria-label="Large"
                        aria-describedby="inputGroup-sizing-sm"
                        />

                        <Dropdown className="dropdown">
                            {openDropdown && loaded ? 
                                <ul className="header__dropdown__content">
                                    <DinoSearchDropdowns 
                                        
                                        dinosauria={dinosauria} 
                                        setOpenDropdown={setOpenDropdown} 
                                        keyword={keyword}
                                        setIdModal={setIdModal} 
                                        setModalIsOpen={setModalIsOpen}
                                    />
                                </ul> : undefined}
                        </Dropdown>
                </InputGroup>

            </div>
        </div>
    ) 
}