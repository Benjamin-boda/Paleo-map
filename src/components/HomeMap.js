import React, {useState, useEffect} from "react";
import { ComposableMap, Geographies, Geography, Marker, ZoomableGroup} from "react-simple-maps";
import world from '../map/world.json';
import {MarkerModal} from "./MarkerModal";
import {DinoFilter} from "./DinoFilter";
import {DinoSearch} from "./DinoSearch";

export const HomeMap = () => {
    const [dinosauria, setDinosauria] = useState([])
    const [idModal, setIdModal] = useState(0)
    const [modalIsOpen, setModalIsOpen] = useState(false)
    const [continentOrCountry, setContinentOrCountry] = useState([])
    const [loaded, setLoaded] = useState(false)
    const [dinosauriaOrderArray, setDinosauriaOrderArray] = useState([])
    const [textId, setTextId] = useState("")

    useEffect(() => {
        async function getPBDBApi() {
            setLoaded(false)
            await fetch(`https://paleobiodb.org/data1.2/specs/list.json?datainfo&rowcount&base_name=Dinosauria&idqual=certain${continentOrCountry}&show=class,abund,coll,coords,loc,geo,acconly&textresult`)
                .then((req) => req.json())
                .then((res) => {
                    setDinosauria(res.records)
                    setDinosauriaOrderArray(res.records)
                    setLoaded(true)
                })
        }

        getPBDBApi()
    }, [continentOrCountry])

    return (
        <div>
            <DinoSearch loaded={loaded} dinosauria={dinosauria} setIdModal={setIdModal} setModalIsOpen={setModalIsOpen}/>

            <div className="map">
                <ComposableMap width={window.innerWidth} height={window.innerHeight * 0.8}>
                    <ZoomableGroup zoom={1} center={[0,0]}>
                        <Geographies geography={world}>
                            {({ geographies }) =>
                                geographies.map(geo => <Geography key={geo.rsmKey} geography={geo} fill="#000000" stroke="#FFF"/>)
                            }
                        </Geographies>
                        {dinosauria.map((dino) =>
                            
                            
                            <Marker key={dino.oid} coordinates={[dino.lng, dino.lat]} 
                                onMouseEnter={() => setTextId(dino.oid)}
                                onMouseLeave={() => setTextId("")}
                                style={{
                                    default: { fill: "#666" },
                                    hover:   { fill: "#ff4500" },
                                    pressed: { fill: "#fff44f" },
                                }}>
                                
                                <circle 
                                    onClick={() => {
                                        setModalIsOpen(!modalIsOpen)
                                        setIdModal(dino.oid)}}
                                        cx="12" cy="10" r="3"  
                                />
                                {textId === dino.oid ? <text className="marker__text" style={{ fontFamily: "OpenSansRegular"}} textAnchor="middle">{dino.gnl}</text> : undefined}
                                {idModal === dino.oid ? <MarkerModal dino={dino} modalIsOpen={modalIsOpen} setModalIsOpen={setModalIsOpen} /> : undefined}
                            </Marker>
                        )}  
                    </ZoomableGroup>   
                </ComposableMap>
            </div>
            
            <DinoFilter
                loaded={loaded}
                continentOrCountry={continentOrCountry}  
                setContinentOrCountry={setContinentOrCountry} 
                dinosauriaOrderArray={dinosauriaOrderArray} 
                setDinosauria={setDinosauria}
            />
        </div>
    )
}