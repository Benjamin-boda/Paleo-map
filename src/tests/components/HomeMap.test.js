import React from "react";
import { HomeMap } from "../../components/HomeMap";
import TestRenderer  from 'react-test-renderer';

let wrapper, tree, instance, select1, select2, select3;

beforeEach(() => {
    wrapper = TestRenderer.create(
        <HomeMap />)
        .toJSON()
});

test("should render homemap correctly", () => {
    expect(wrapper).toMatchSnapshot();
});