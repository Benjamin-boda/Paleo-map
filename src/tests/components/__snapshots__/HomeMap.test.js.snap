// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render homemap correctly 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="header__content"
    >
      <img
        className="header__logo"
      />
      <h1
        className="header__text"
      >
        PALEO MAP
      </h1>
      <div
        className="header__input input-group input-group-lg"
        onChange={[Function]}
        placeholder="Search a specimen..."
        type="text"
        value=""
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="InputGroup__Text input-group-text"
            id="inputGroup-sizing-lg"
          >
            Search a specimen
          </span>
        </div>
        <input
          aria-describedby="inputGroup-sizing-sm"
          aria-label="Large"
          className="form-control"
        />
        <div
          className="dropdown dropdown"
          onKeyDown={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="map"
  >
    <svg
      className="rsm-svg "
      viewBox="0 0 1024 614.4000000000001"
    >
      <g>
        <rect
          fill="transparent"
          height={614.4000000000001}
          width={1024}
        />
        <g
          className="rsm-zoomable-group undefined"
          transform="translate(0 0) scale(1)"
        >
          <g
            className="rsm-geographies "
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className="filters__background"
  >
    <div
      className="filters"
    >
      <select
        className="filters__select form-control"
        id="continent"
        onChange={[Function]}
      >
        <option
          value=""
        >
          --Filter by continent--
        </option>
        <option
          value="&cc=ATA"
        >
          Antarctica
        </option>
        <option
          value="&cc=AFR"
        >
          Africa
        </option>
        <option
          value="&cc=ASI"
        >
          Asia
        </option>
        <option
          value="&cc=EUR"
        >
          Europe
        </option>
        <option
          value="&cc=NOA"
        >
          North America
        </option>
        <option
          value="&cc=OCE"
        >
          Oceania
        </option>
        <option
          value="&cc=SOA"
        >
          South America
        </option>
      </select>
      <select
        className="filters__select form-control"
        onChange={[Function]}
      >
        <option
          value=""
        >
          --Filter by country--
        </option>
        <option
          value="&cc=AF"
        >
          Afghanistan
        </option>
        <option
          value="&cc=AL"
        >
          Albania
        </option>
        <option
          value="&cc=AQ"
        >
          Antarctica (the territory South of 60 deg S)
        </option>
        <option
          value="&cc=DZ"
        >
          Algeria
        </option>
        <option
          value="&cc=AS"
        >
          American Samoa
        </option>
        <option
          value="&cc=AD"
        >
          Andorra
        </option>
        <option
          value="&cc=AO"
        >
          Angola
        </option>
        <option
          value="&cc=AG"
        >
          Antigua and Barbuda
        </option>
        <option
          value="&cc=AZ"
        >
          Azerbaijan
        </option>
        <option
          value="&cc=AZ"
        >
          Azerbaijan
        </option>
        <option
          value="&cc=AR"
        >
          Argentina
        </option>
        <option
          value="&cc=AU"
        >
          Australia
        </option>
        <option
          value="&cc=AT"
        >
          Austria
        </option>
        <option
          value="&cc=BS"
        >
          Bahamas
        </option>
        <option
          value="&cc=BH"
        >
          Bahrain
        </option>
        <option
          value="&cc=BD"
        >
          Bangladesh
        </option>
        <option
          value="&cc=AM"
        >
          Armenia
        </option>
        <option
          value="&cc=AM"
        >
          Armenia
        </option>
        <option
          value="&cc=BB"
        >
          Barbados
        </option>
        <option
          value="&cc=BE"
        >
          Belgium
        </option>
        <option
          value="&cc=BM"
        >
          Bermuda
        </option>
        <option
          value="&cc=BT"
        >
          Bhutan
        </option>
        <option
          value="&cc=BO"
        >
          Bolivia
        </option>
        <option
          value="&cc=BA"
        >
          Bosnia and Herzegovina
        </option>
        <option
          value="&cc=BW"
        >
          Botswana
        </option>
        <option
          value="&cc=BV"
        >
          Bouvet Island (Bouvetoya)
        </option>
        <option
          value="&cc=BR"
        >
          Brazil
        </option>
        <option
          value="&cc=BZ"
        >
          Belize
        </option>
        <option
          value="&cc=IO"
        >
          British Indian Ocean Territory (Chagos Archipelago)
        </option>
        <option
          value="&cc=SB"
        >
          Solomon Islands
        </option>
        <option
          value="&cc=VG"
        >
          British Virgin Islands
        </option>
        <option
          value="&cc=BN"
        >
          Brunei Darussalam
        </option>
        <option
          value="&cc=BG"
        >
          Bulgaria
        </option>
        <option
          value="&cc=MM"
        >
          Myanmar
        </option>
        <option
          value="&cc=BI"
        >
          Burundi
        </option>
        <option
          value="&cc=BY"
        >
          Belarus
        </option>
        <option
          value="&cc=KH"
        >
          Cambodia
        </option>
        <option
          value="&cc=CM"
        >
          Cameroon
        </option>
        <option
          value="&cc=CA"
        >
          Canada
        </option>
        <option
          value="&cc=CV"
        >
          Cape Verde
        </option>
        <option
          value="&cc=KY"
        >
          Cayman Islands
        </option>
        <option
          value="&cc=CF"
        >
          Central African Republic
        </option>
        <option
          value="&cc=LK"
        >
          Sri Lanka
        </option>
        <option
          value="&cc=TD"
        >
          Chad
        </option>
        <option
          value="&cc=CL"
        >
          Chile
        </option>
        <option
          value="&cc=CN"
        >
          China
        </option>
        <option
          value="&cc=TW"
        >
          Taiwan
        </option>
        <option
          value="&cc=CX"
        >
          Christmas Island
        </option>
        <option
          value="&cc=CC"
        >
          Cocos (Keeling) Islands
        </option>
        <option
          value="&cc=CO"
        >
          Colombia
        </option>
        <option
          value="&cc=KM"
        >
          Comoros
        </option>
        <option
          value="&cc=YT"
        >
          Mayotte
        </option>
        <option
          value="&cc=CG"
        >
          Congo
        </option>
        <option
          value="&cc=CD"
        >
          Congo
        </option>
        <option
          value="&cc=CK"
        >
          Cook Islands
        </option>
        <option
          value="&cc=CR"
        >
          Costa Rica
        </option>
        <option
          value="&cc=HR"
        >
          Croatia
        </option>
        <option
          value="&cc=CU"
        >
          Cuba
        </option>
        <option
          value="&cc=CY"
        >
          Cyprus
        </option>
        <option
          value="&cc=CY"
        >
          Cyprus
        </option>
        <option
          value="&cc=CZ"
        >
          Czech Republic
        </option>
        <option
          value="&cc=BJ"
        >
          Benin
        </option>
        <option
          value="&cc=DK"
        >
          Denmark
        </option>
        <option
          value="&cc=DM"
        >
          Dominica
        </option>
        <option
          value="&cc=DO"
        >
          Dominican Republic
        </option>
        <option
          value="&cc=EC"
        >
          Ecuador
        </option>
        <option
          value="&cc=SV"
        >
          El Salvador
        </option>
        <option
          value="&cc=GQ"
        >
          Equatorial Guinea
        </option>
        <option
          value="&cc=ET"
        >
          Ethiopia
        </option>
        <option
          value="&cc=ER"
        >
          Eritrea
        </option>
        <option
          value="&cc=EE"
        >
          Estonia
        </option>
        <option
          value="&cc=FO"
        >
          Faroe Islands
        </option>
        <option
          value="&cc=FK"
        >
          Falkland Islands (Malvinas)
        </option>
        <option
          value="&cc=GS"
        >
          South Georgia and the South Sandwich Islands
        </option>
        <option
          value="&cc=FJ"
        >
          Fiji
        </option>
        <option
          value="&cc=FI"
        >
          Finland
        </option>
        <option
          value="&cc=AX"
        >
          Ã…land Islands
        </option>
        <option
          value="&cc=FR"
        >
          France
        </option>
        <option
          value="&cc=GF"
        >
          French Guiana
        </option>
        <option
          value="&cc=PF"
        >
          French Polynesia
        </option>
        <option
          value="&cc=TF"
        >
          French Southern Territories
        </option>
        <option
          value="&cc=DJ"
        >
          Djibouti
        </option>
        <option
          value="&cc=GA"
        >
          Gabon
        </option>
        <option
          value="&cc=GE"
        >
          Georgia
        </option>
        <option
          value="&cc=GE"
        >
          Georgia
        </option>
        <option
          value="&cc=GM"
        >
          Gambia
        </option>
        <option
          value="&cc=PS"
        >
          Palestinian Territory
        </option>
        <option
          value="&cc=DE"
        >
          Germany
        </option>
        <option
          value="&cc=GH"
        >
          Ghana
        </option>
        <option
          value="&cc=GI"
        >
          Gibraltar
        </option>
        <option
          value="&cc=KI"
        >
          Kiribati
        </option>
        <option
          value="&cc=GR"
        >
          Greece
        </option>
        <option
          value="&cc=GL"
        >
          Greenland
        </option>
        <option
          value="&cc=GD"
        >
          Grenada
        </option>
        <option
          value="&cc=GP"
        >
          Guadeloupe
        </option>
        <option
          value="&cc=GU"
        >
          Guam
        </option>
        <option
          value="&cc=GT"
        >
          Guatemala
        </option>
        <option
          value="&cc=GN"
        >
          Guinea
        </option>
        <option
          value="&cc=GY"
        >
          Guyana
        </option>
        <option
          value="&cc=HT"
        >
          Haiti
        </option>
        <option
          value="&cc=HM"
        >
          Heard Island and McDonald Islands
        </option>
        <option
          value="&cc=VA"
        >
          Holy See (Vatican City State)
        </option>
        <option
          value="&cc=HN"
        >
          Honduras
        </option>
        <option
          value="&cc=HK"
        >
          Hong Kong
        </option>
        <option
          value="&cc=HU"
        >
          Hungary
        </option>
        <option
          value="&cc=IS"
        >
          Iceland
        </option>
        <option
          value="&cc=IN"
        >
          India
        </option>
        <option
          value="&cc=ID"
        >
          Indonesia
        </option>
        <option
          value="&cc=IR"
        >
          Iran
        </option>
        <option
          value="&cc=IQ"
        >
          Iraq
        </option>
        <option
          value="&cc=IE"
        >
          Ireland
        </option>
        <option
          value="&cc=IL"
        >
          Israel
        </option>
        <option
          value="&cc=IT"
        >
          Italy
        </option>
        <option
          value="&cc=CI"
        >
          Cote d'Ivoire
        </option>
        <option
          value="&cc=JM"
        >
          Jamaica
        </option>
        <option
          value="&cc=JP"
        >
          Japan
        </option>
        <option
          value="&cc=KZ"
        >
          Kazakhstan
        </option>
        <option
          value="&cc=KZ"
        >
          Kazakhstan
        </option>
        <option
          value="&cc=JO"
        >
          Jordan
        </option>
        <option
          value="&cc=KE"
        >
          Kenya
        </option>
        <option
          value="&cc=KP"
        >
          Korea
        </option>
        <option
          value="&cc=KR"
        >
          Korea
        </option>
        <option
          value="&cc=KW"
        >
          Kuwait
        </option>
        <option
          value="&cc=KG"
        >
          Kyrgyz Republic
        </option>
        <option
          value="&cc=LA"
        >
          Lao People's Democratic Republic
        </option>
        <option
          value="&cc=LB"
        >
          Lebanon
        </option>
        <option
          value="&cc=LS"
        >
          Lesotho
        </option>
        <option
          value="&cc=LV"
        >
          Latvia
        </option>
        <option
          value="&cc=LR"
        >
          Liberia
        </option>
        <option
          value="&cc=LY"
        >
          Libyan Arab Jamahiriya
        </option>
        <option
          value="&cc=LI"
        >
          Liechtenstein
        </option>
        <option
          value="&cc=LT"
        >
          Lithuania
        </option>
        <option
          value="&cc=LU"
        >
          Luxembourg
        </option>
        <option
          value="&cc=MO"
        >
          Macao
        </option>
        <option
          value="&cc=MG"
        >
          Madagascar
        </option>
        <option
          value="&cc=MW"
        >
          Malawi
        </option>
        <option
          value="&cc=MY"
        >
          Malaysia
        </option>
        <option
          value="&cc=MV"
        >
          Maldives
        </option>
        <option
          value="&cc=ML"
        >
          Mali
        </option>
        <option
          value="&cc=MT"
        >
          Malta
        </option>
        <option
          value="&cc=MQ"
        >
          Martinique
        </option>
        <option
          value="&cc=MR"
        >
          Mauritania
        </option>
        <option
          value="&cc=MU"
        >
          Mauritius
        </option>
        <option
          value="&cc=MX"
        >
          Mexico
        </option>
        <option
          value="&cc=MC"
        >
          Monaco
        </option>
        <option
          value="&cc=MN"
        >
          Mongolia
        </option>
        <option
          value="&cc=MD"
        >
          Moldova
        </option>
        <option
          value="&cc=ME"
        >
          Montenegro
        </option>
        <option
          value="&cc=MS"
        >
          Montserrat
        </option>
        <option
          value="&cc=MA"
        >
          Morocco
        </option>
        <option
          value="&cc=MZ"
        >
          Mozambique
        </option>
        <option
          value="&cc=OM"
        >
          Oman
        </option>
        <option
          value="&cc=NA"
        >
          Namibia
        </option>
        <option
          value="&cc=NR"
        >
          Nauru
        </option>
        <option
          value="&cc=NP"
        >
          Nepal
        </option>
        <option
          value="&cc=NL"
        >
          Netherlands
        </option>
        <option
          value="&cc=AN"
        >
          Netherlands Antilles
        </option>
        <option
          value="&cc=CW"
        >
          CuraÃ§ao
        </option>
        <option
          value="&cc=AW"
        >
          Aruba
        </option>
        <option
          value="&cc=SX"
        >
          Sint Maarten (Netherlands)
        </option>
        <option
          value="&cc=BQ"
        >
          Bonaire
        </option>
        <option
          value="&cc=NC"
        >
          New Caledonia
        </option>
        <option
          value="&cc=VU"
        >
          Vanuatu
        </option>
        <option
          value="&cc=NZ"
        >
          New Zealand
        </option>
        <option
          value="&cc=NI"
        >
          Nicaragua
        </option>
        <option
          value="&cc=NE"
        >
          Niger
        </option>
        <option
          value="&cc=NG"
        >
          Nigeria
        </option>
        <option
          value="&cc=NU"
        >
          Niue
        </option>
        <option
          value="&cc=NF"
        >
          Norfolk Island
        </option>
        <option
          value="&cc=NO"
        >
          Norway
        </option>
        <option
          value="&cc=MP"
        >
          Northern Mariana Islands
        </option>
        <option
          value="&cc=UM"
        >
          United States Minor Outlying Islands
        </option>
        <option
          value="&cc=UM"
        >
          United States Minor Outlying Islands
        </option>
        <option
          value="&cc=FM"
        >
          Micronesia
        </option>
        <option
          value="&cc=MH"
        >
          Marshall Islands
        </option>
        <option
          value="&cc=PW"
        >
          Palau
        </option>
        <option
          value="&cc=PK"
        >
          Pakistan
        </option>
        <option
          value="&cc=PA"
        >
          Panama
        </option>
        <option
          value="&cc=PG"
        >
          Papua New Guinea
        </option>
        <option
          value="&cc=PY"
        >
          Paraguay
        </option>
        <option
          value="&cc=PE"
        >
          Peru
        </option>
        <option
          value="&cc=PH"
        >
          Philippines
        </option>
        <option
          value="&cc=PN"
        >
          Pitcairn Islands
        </option>
        <option
          value="&cc=PL"
        >
          Poland
        </option>
        <option
          value="&cc=PT"
        >
          Portugal
        </option>
        <option
          value="&cc=GW"
        >
          Guinea-Bissau
        </option>
        <option
          value="&cc=TL"
        >
          Timor-Leste
        </option>
        <option
          value="&cc=PR"
        >
          Puerto Rico
        </option>
        <option
          value="&cc=QA"
        >
          Qatar
        </option>
        <option
          value="&cc=RE"
        >
          Reunion
        </option>
        <option
          value="&cc=RO"
        >
          Romania
        </option>
        <option
          value="&cc=RU"
        >
          Russian Federation
        </option>
        <option
          value="&cc=RU"
        >
          Russian Federation
        </option>
        <option
          value="&cc=RW"
        >
          Rwanda
        </option>
        <option
          value="&cc=BL"
        >
          Saint Barthelemy
        </option>
        <option
          value="&cc=SH"
        >
          Saint Helena
        </option>
        <option
          value="&cc=KN"
        >
          Saint Kitts and Nevis
        </option>
        <option
          value="&cc=AI"
        >
          Anguilla
        </option>
        <option
          value="&cc=LC"
        >
          Saint Lucia
        </option>
        <option
          value="&cc=MF"
        >
          Saint Martin
        </option>
        <option
          value="&cc=PM"
        >
          Saint Pierre and Miquelon
        </option>
        <option
          value="&cc=VC"
        >
          Saint Vincent and the Grenadines
        </option>
        <option
          value="&cc=SM"
        >
          San Marino
        </option>
        <option
          value="&cc=ST"
        >
          Sao Tome and Principe
        </option>
        <option
          value="&cc=SA"
        >
          Saudi Arabia
        </option>
        <option
          value="&cc=SN"
        >
          Senegal
        </option>
        <option
          value="&cc=RS"
        >
          Serbia
        </option>
        <option
          value="&cc=SC"
        >
          Seychelles
        </option>
        <option
          value="&cc=SL"
        >
          Sierra Leone
        </option>
        <option
          value="&cc=SG"
        >
          Singapore
        </option>
        <option
          value="&cc=SK"
        >
          Slovakia (Slovak Republic)
        </option>
        <option
          value="&cc=VN"
        >
          Vietnam
        </option>
        <option
          value="&cc=SI"
        >
          Slovenia
        </option>
        <option
          value="&cc=SO"
        >
          Somalia
        </option>
        <option
          value="&cc=ZA"
        >
          South Africa
        </option>
        <option
          value="&cc=ZW"
        >
          Zimbabwe
        </option>
        <option
          value="&cc=ES"
        >
          Spain
        </option>
        <option
          value="&cc=SS"
        >
          South Sudan
        </option>
        <option
          value="&cc=EH"
        >
          Western Sahara
        </option>
        <option
          value="&cc=SD"
        >
          Sudan
        </option>
        <option
          value="&cc=SR"
        >
          Suriname
        </option>
        <option
          value="&cc=SJ"
        >
          Svalbard & Jan Mayen Islands
        </option>
        <option
          value="&cc=SZ"
        >
          Swaziland
        </option>
        <option
          value="&cc=SE"
        >
          Sweden
        </option>
        <option
          value="&cc=CH"
        >
          Switzerland
        </option>
        <option
          value="&cc=SY"
        >
          Syrian Arab Republic
        </option>
        <option
          value="&cc=TJ"
        >
          Tajikistan
        </option>
        <option
          value="&cc=TH"
        >
          Thailand
        </option>
        <option
          value="&cc=TG"
        >
          Togo
        </option>
        <option
          value="&cc=TK"
        >
          Tokelau
        </option>
        <option
          value="&cc=TO"
        >
          Tonga
        </option>
        <option
          value="&cc=TT"
        >
          Trinidad and Tobago
        </option>
        <option
          value="&cc=AE"
        >
          United Arab Emirates
        </option>
        <option
          value="&cc=TN"
        >
          Tunisia
        </option>
        <option
          value="&cc=TR"
        >
          Turkey
        </option>
        <option
          value="&cc=TR"
        >
          Turkey
        </option>
        <option
          value="&cc=TM"
        >
          Turkmenistan
        </option>
        <option
          value="&cc=TC"
        >
          Turks and Caicos Islands
        </option>
        <option
          value="&cc=TV"
        >
          Tuvalu
        </option>
        <option
          value="&cc=UG"
        >
          Uganda
        </option>
        <option
          value="&cc=UA"
        >
          Ukraine
        </option>
        <option
          value="&cc=MK"
        >
          Macedonia
        </option>
        <option
          value="&cc=EG"
        >
          Egypt
        </option>
        <option
          value="&cc=GB"
        >
          United Kingdom of Great Britain & Northern Ireland
        </option>
        <option
          value="&cc=GG"
        >
          Guernsey
        </option>
        <option
          value="&cc=JE"
        >
          Jersey
        </option>
        <option
          value="&cc=IM"
        >
          Isle of Man
        </option>
        <option
          value="&cc=TZ"
        >
          Tanzania
        </option>
        <option
          value="&cc=US"
        >
          United States of America
        </option>
        <option
          value="&cc=VI"
        >
          United States Virgin Islands
        </option>
        <option
          value="&cc=BF"
        >
          Burkina Faso
        </option>
        <option
          value="&cc=UY"
        >
          Uruguay
        </option>
        <option
          value="&cc=UZ"
        >
          Uzbekistan
        </option>
        <option
          value="&cc=VE"
        >
          Venezuela
        </option>
        <option
          value="&cc=WF"
        >
          Wallis and Futuna
        </option>
        <option
          value="&cc=WS"
        >
          Samoa
        </option>
        <option
          value="&cc=YE"
        >
          Yemen
        </option>
        <option
          value="&cc=ZM"
        >
          Zambia
        </option>
        <option
          value="&cc=XX"
        >
          Disputed Territory
        </option>
        <option
          value="&cc=XE"
        >
          Iraq-Saudi Arabia Neutral Zone
        </option>
        <option
          value="&cc=XD"
        >
          United Nations Neutral Zone
        </option>
        <option
          value="&cc=XS"
        >
          Spratly Islands
        </option>
      </select>
      <select
        className="filters__select form-control"
        onChange={[Function]}
      >
        <option
          value=""
        >
          --Filter by order--
        </option>
        <option
          value="Saurischia"
        >
          Saurischia
        </option>
        <option
          value="Ornithischia"
        >
          Ornithischia
        </option>
        <option
          value="Aves"
        >
          Aves
        </option>
      </select>
    </div>
  </div>
</div>
`;
